### ISSUE TITLE
iOS: Ti.Media.openPhotoGallery does not respect Selected Photos permission

### ISSUE THREAD

**Issue by phillipmovista** [appcelerator/titanium_mobile#13277]
### I have searched and made sure there are no existing issues for the issue I am filing

- [X] I have searched the existing issues

### Description

When we use `Ti.Media.openPhotoGallery` for selecting photos in our application, after selecting iOS Photo Permissions to only allow access to "selected photos", the `openPhotoGallery` function still gives the app access to all photos.

### Expected Behavior

After our code calls `Ti.Media.openPhotoGallery` like https://docs.axway.com/bundle/Titanium_SDK_allOS_en/page/camera_and_photo_gallery_apis.html#CameraandPhotoGalleryAPIs-Gallery, we do get a popup allowing us to select the photo permission and after that, a gallery pops up.

If I use the Selected Photos option, I can see later in Settings that the limited number of photos are showing they are accessible, but the app still allows all photos.

Details from iOS and this new security is: https://developer.apple.com/documentation/photokit/delivering_an_enhanced_privacy_experience_in_your_photos_app

### Actual behavior

This shows the permissions in Settings about which photos should be available:
<img width="424" alt="Screen Shot 2022-02-21 at 12 22 54 PM" src="https://user-images.githubusercontent.com/51764912/155022618-b5aabafa-870f-4fe0-ae68-67b472a58ae4.png">

When I then go into the app and run the code that does `Ti.Media.openPhotoGallery` I see these photos as options to select:
<img width="430" alt="Screen Shot 2022-02-21 at 12 23 08 PM" src="https://user-images.githubusercontent.com/51764912/155022670-ee3df8c0-d53e-407d-a09a-e5abab5787fe.png">

I would expect to see only the 3 photos I gave permission to.

### Reproducible sample

We have some other code around this for the callbacks (which work as expected) but this is the code with some extra `console` statements to show how we are opening this option.

```
Ti.Media.openPhotoGallery({
	success: (e) => {
		for (var i = 0; i < e.images.length; i++) {
			console.error(e.images[i]);

			selectedItems.push({
				media: e.images[i].media
			});
		}
		done();
	},
	cancel: () => {
		cancel();
	},
	error: (error) => {
		console.error('Photo gallery error: ', error);
		cancel();
	},
	autohide: true,
	allowEditing: false,
	allowMultiple: true,
	selectionLimit: args.max,
	mediaTypes: [Ti.Media.MEDIA_TYPE_PHOTO]
});
```

### Steps to reproduce

In our case, the reproduction is that once the dialog box opens the first time asking which photo permissions to use, choose "Selected Photos" and just select a few photos. You can then verify that inside Settings to show it is only allowing access to some photos.

In the application, you should see all photos available.

### Platform

iOS

### SDK version you are using

10.1.1.GA

**Comment by phillipmovista** [comment]
I am going to see about using `ti.imagepicker` for now (https://github.com/hansemannn/titanium-imagepicker) because it does seem to handle the permissions properly.

**Comment by hansemannn** [comment]
The reason why it shows all photos is because of the `PHPickerViewController` being used on iOS 13+, which is a sandbox-based API that does not share the photos with the app unless selected. Therefore, it's at least not that critical. It would be interesting to know what exactly you would expect instead.

**Comment by hansemannn** [comment]
Yep, the reason for that is that the `ti.imagepicker` creates a custom UI (using the `YPImagePicker` library), while `Ti.Media.openPhotoGallery` uses the native `PHPickerViewController` class (system iOS class) directly. Thats where the difference comes from. If you want only the ones that you selected, either use `[allowMultiple: 1](https://titaniumsdk.com/api/structs/photogalleryoptionstype.html#allowMultiple)` (which will use the "old" dialog that only returns granted photos, or use the ti.imagepicker UI with a custom UI. That's basically what's possible natively.

**Comment by phillipmovista** [comment]
Thank you. I am using the `ti.imagepicker` now, but even with `[allowMultiple: 1]` option, I still found it was returning me more than the ones with permission for the app to select. To me that seems like a bug in the `Ti.Media` layer. Is that right?

**Comment by hansemannn** [comment]
Sorry, I mean `allowMultiple: false`. But happy it works with ti.imagepicker for now!

**Comment by phillipmovista** [comment]
Oh, so if we need a way to select multiple at a time, we should go with `ti.imagepicker` and if we use `openPhotoGallery`, it only will respect the OS permissions 100% with 1 photo at a time?

My recommendation would still be to treat this as a bug since that is hard to see happening and I would think the expectation is always to follow the OS permissions.
